lib/reverse_words.rb

```diff
@@ -1,6 +1,31 @@
+require 'pry'
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: On - linear
+# Space complexity: O - no additional space taken up, temporary values to move things around. 
 def reverse_words(my_words)
-  raise NotImplementedError
-end
+  return my_words if my_words == nil || my_words.length == 0
+  
+  i = 0
+  l = my_words.length
+  start_char = 0
+  finish_char = 0
+  
+  while i < l 
```

Comment:
I'd suggest renaming `l` to `length` for clarity. Too many single letter variables (especially when white boarding) can lead to added stress.